### PR TITLE
Speed up G35

### PR DESCRIPTION
**Pull request by qwewer0**
### Description

Check if all axis trusted and home only Z axis when X and Y is trusted, otherwise all axes.

### Requirements

`ASSISTED_TRAMMING` enabled

### Benefits

G35 tramming now takes less time when it needs to be done multiple times in a row.

### Configurations

None

### Related Issues

None

### CODE REVIEW

**Comment by InsanityAutomation** [comment]
If any of the corners move and homing is done by a probe in the center, the zero point will have changed. Even if doing G35 repeatedly it may keep the delta to each other however the first point will be an irrelevant number with no real point of reference.

I'd argue it should stay and even at the loosest I'd consider, allow it to be overriden with an argument to G35

**Comment by thinkyhead** [comment]
It seems like it could re-home Z at the end before re-enabling leveling, but maybe re-homing Z is undesirable…?

**Comment by InsanityAutomation** [comment]
Yeah, enabling leveling doesn't seem to make sense here considering by intent it's invalid now. I would be invalidating the mesh if I couldn't tilt it with UBL G29J.

**Comment by qwewer0** [comment]
If we add a z only homing at the start which will home everything if x and y too is untrusted and at the end we only make z untrusted, then G35 is still faster, but for further operations Z will need homing for safety, what do we think?

**Comment by qwewer0** [comment]
> It seems like it could re-home Z at the end before re-enabling leveling, but maybe re-homing Z is undesirable…?

**Comment by InsanityAutomation** [comment]
The home if needed at the beginning is fine, just at the end it needs to re-check Z and the mesh should be invalidated if not UBL.
Also, the home if needed naming is a little misleading. Id expect it to home only the axis not trusted, where it currently homes all if anything isnt trusted. Making a function for it seems overkill if it doesnt add at minimum that extra bit of complexity to it, otherwise its replacing 1 line with 1 shorter line.

**Comment by qwewer0** [comment]
> at the end it needs to re-check Z

Not sure how can the firmware know when is the end of the full G35 procedure with the knobs adjusted, that is why I suggested the z home and z untrust option.

**Comment by thinkyhead** [comment]
I dropped by to read the comments. I'll mull it over and come back later.

File: Marlin/src/gcode/bedlevel/G35.cpp [hunk 1]
```diff
@@ -91,8 +91,8 @@ void GcodeSuite::G35() {
   // Disable duplication mode on homing
   TERN_(HAS_DUPLICATION_MODE, set_duplication_enabled(false));
 
-  // Home all before this procedure
-  home_all_axes();
+  // Home only Z axis when X and Y is trusted, otherwise all axes, if needed before this procedure
+  if (!all_axes_trusted()) queue.inject_P(PSTR("G28Z"));
```
Comment (qwewer0) [inline Marlin/src/gcode/bedlevel/G35.cpp:95]: Not sure if this is the right way to home Z, so I'm up for the corrections.

Comment (thinkyhead) [reply Marlin/src/gcode/bedlevel/G35.cpp:95]: From within a G-code handler you can always use `process_subcommands_now` and `process_subcommands_now_P`.

**Pull request merged**